src/shadowbox/server/main.ts

```diff
@@ -160,12 +136,13 @@ async function main() {
       getPersistentFilename('prometheus/config.yml'), prometheusConfigJson);
 
   const prometheusClient = new PrometheusClient(`http://${prometheusLocation}`);
+  if (!serverConfig.data().portForNewAccessKeys) {
+    serverConfig.data().portForNewAccessKeys = await portProvider.reserveNewPort();
+  }
+  serverConfig.write();
```

Comment:
We only need to save the config when assigning the default port. Please move to the if statement.

Reply:
Done